exec/java-exec/src/test/java/org/apache/drill/exec/resourcemgr/TestRMConfigLoad.java

```diff
@@ -0,0 +1,114 @@
+        8 * 1024L, poolTree.getRootPool().getMaxQueryMemoryPerNode());
+      assertEquals("queue_selection_policy in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.ROOT_POOL_DEFAULT_QUEUE_SELECTION_POLICY,
+        poolTree.getSelectionPolicyInUse().getSelectionPolicyName());
+      assertEquals("memory share of root pool in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.ROOT_POOL_DEFAULT_MEMORY_PERCENT, poolTree.getResourceShare(), 0);
+
+      final QueryQueueConfig defaultQueue = poolTree.getRootPool().getQueuryQueue();
+      assertEquals("max_admissible in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.MAX_ADMISSIBLE_DEFAULT, defaultQueue.getMaxAdmissibleQueries());
+      assertEquals("max_waiting in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.MAX_WAITING_DEFAULT, defaultQueue.getMaxWaitingQueries());
+      assertEquals("max_wait_timeout in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.MAX_WAIT_TIMEOUT_IN_MS_DEFAULT, defaultQueue.getWaitTimeoutInMs());
+      assertEquals("wait_for_preferred_nodes in drill-rm-default is not configured with expected default value",
+        RMCommonDefaults.WAIT_FOR_PREFERRED_NODES_DEFAULT, defaultQueue.waitForPreferredNodes());
+    }
+  }
+
+  @Test
+  public void testDefaultRMWithLocalCoordinatorAndRMEnabled() throws Exception {
+    ClusterFixtureBuilder fixtureBuilder = ClusterFixture.builder(dirTestWatcher)
+      .configProperty(ExecConstants.RM_ENABLED, true);
+
+    try (ClusterFixture cluster = fixtureBuilder.build()) {
+      ResourceManager resourceManager = cluster.drillbit().getContext().getResourceManager();
+      assertTrue(resourceManager instanceof DefaultResourceManager);
+    }
+  }
+
+  @Test
+  public void testDefaultRMWithLocalCoordinatorAndRMDisabled() throws Exception {
+    ClusterFixtureBuilder fixtureBuilder = ClusterFixture.builder(dirTestWatcher)
+      .configProperty(ExecConstants.RM_ENABLED, false);
+
+    try (ClusterFixture cluster = fixtureBuilder.build()) {
+      ResourceManager resourceManager = cluster.drillbit().getContext().getResourceManager();
+      assertTrue(resourceManager instanceof DefaultResourceManager);
+    }
+  }
+
+  @Test
+  public void testDefaultRMOnlyRMDisabled() throws Exception {
+    ClusterFixtureBuilder fixtureBuilder = ClusterFixture.builder(dirTestWatcher)
+      .configProperty(ExecConstants.RM_ENABLED, false)
+      .configProperty(ExecConstants.DRILL_PORT_HUNT, true)
+      .withLocalZk();
+
+    try (ClusterFixture cluster = fixtureBuilder.build()) {
+      ResourceManager resourceManager = cluster.drillbit().getContext().getResourceManager();
+      assertTrue(resourceManager instanceof DefaultResourceManager);
+    }
```

Comment:
All the tests are positive tests in that ResourceManager is always instance of DefaultResourceManager. Are you planning to add more tests when the tests are enabled?